### ISSUE TITLE
An error was reported while downloading the dataset.

### ISSUE THREAD

**Issue by theneao** [mathpluscode/ImgX-DiffSeg#3]
````
Extraction completed...: 0 file [1:26:56, ? file/s]1:26:56<00:00, 5216.07s/ url]
Dl Size...:   6%|█▎                   | 189/3135 [1:26:56<22:35:04, 27.60s/ MiB]
Dl Completed...: 100%|██████████████████████| 1/1 [1:26:56<00:00, 5216.08s/ url]
Traceback (most recent call last):  

  File "/home/mhb/anaconda3/envs/imgx/bin/tfds", line 8, in <module>
    sys.exit(launch_cli())
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/scripts/cli/main.py", line 109, in launch_cli
    app.run(main, flags_parser=_parse_flags)
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/absl/app.py", line 308, in run
    _run_main(main, args)
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/absl/app.py", line 254, in _run_main
    sys.exit(main(argv))
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/scripts/cli/main.py", line 104, in main
    args.subparser_fn(args)
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/scripts/cli/build.py", line 274, in _build_datasets
    _download_and_prepare(args, builder)
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/scripts/cli/build.py", line 510, in _download_and_prepare
    builder.download_and_prepare(
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/core/logging/__init__.py", line 169, in __call__
    return function(*args, **kwargs)
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/core/dataset_builder.py", line 640, in download_and_prepare
    self._download_and_prepare(
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/core/dataset_builder.py", line 1448, in _download_and_prepare
    split_generators = self._split_generators(  # pylint: disable=unexpected-keyword-arg
  File "/media/mhb/jxp01/workspace/xq/ImgX-DiffSeg/imgx/datasets/male_pelvic_mr/male_pelvic_mr_dataset_builder.py", line 114, in _split_generators
    zip_dir = dl_manager.download_and_extract(
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/core/download/download_manager.py", line 686, in download_and_extract
    return _map_promise(self._download_extract, url_or_urls)
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/core/download/download_manager.py", line 829, in _map_promise
    res = tree_utils.map_structure(
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tree/__init__.py", line 435, in map_structure
    [func(*args) for args in zip(*map(flatten, structures))])
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tree/__init__.py", line 435, in <listcomp>
    [func(*args) for args in zip(*map(flatten, structures))])
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/core/download/download_manager.py", line 830, in <lambda>
    lambda p: p.get(), all_promises
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/promise/promise.py", line 512, in get
    return self._target_settled_value(_raise=True)
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/promise/promise.py", line 516, in _target_settled_value
    return self._target()._settled_value(_raise)
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/promise/promise.py", line 226, in _settled_value
    reraise(type(raise_val), raise_val, self._traceback)
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/six.py", line 719, in reraise
    raise value
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/promise/promise.py", line 87, in try_catch
    return (handler(*args, **kwargs), None)
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/core/download/download_manager.py", line 406, in <lambda>
    lambda dl_result: self._register_or_validate_checksums(  # pylint: disable=g-long-lambda
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/core/download/download_manager.py", line 471, in _register_or_validate_checksums
    return self._rename_and_get_final_dl_path(
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/core/download/download_manager.py", line 510, in _rename_and_get_final_dl_path
    resource_lib.write_info_file(
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/core/utils/py_utils.py", line 496, in lock_decorated
    return fn(*args, **kwargs)
  File "/home/mhb/anaconda3/envs/imgx/lib/python3.8/site-packages/tensorflow_datasets/core/download/resource.py", line 267, in write_info_file
    raise ValueError(
ValueError: File info /home/mhb/tensorflow_datasets/downloads/zenodo.org_record_7013610_files_dataW0mCI6aH_V-TdeDbM4TdKelNcJ5ZxbAi5isebqCnMr0.zip.INFO contains a different checksum that the downloaded one: Stored: {'checksum': 'c26ec704090bb4a705e0903a05a3d9ac7d7702284eda0f27f19dc17909bd1ead', 'filename': 'data.zip', 'size': 5880838}; Expected: {'size': 189.95 MiB, 'checksum': 'd9270ea4881c4e1002bc2b181213f16ce3ad5f1e93aad17e160246cc0d717d53', 'filename': 'data.zip'}`
````

An error was reported while downloading the dataset. It seems that the download has not been completed. "My download here is relatively slow, and I cannot continue downloading after an interruption, so I can only download it again.". 
_What changes have occurred to the data processed by TFPS, which is convenient for me to download and modify the original dataset, or if it is convenient for you to upload the processed dataset to a specific website_

**Comment by theneao** [comment]
I am very excited about the application of diffusion models in semantic segmentation tasks. After carefully reading your paper, I feel that your work is very meaningful. However, it is also a pity that you have not surpassed the current SOTA in 2D. I also hope to promote and improve your method to achieve greater improvement in the field of 2D image segmentation. Looking forward to your reply!!!

**Comment by mathpluscode** [comment]
Hi @theneao, I am not familiar with the error you raised. I wonder which version of `tfds` you are using? This issue may be helpful https://github.com/tensorflow/datasets/issues/282?

Otherwise, there was a small typo in the readme, see #2. We have fixed it and hope it can resolve the error.

Thanks for your interest :)

**Comment by theneao** [comment]
I have updated some of the relevant libraries for tfds (previously these are also the latest) that are not working. The main reason may be network issues, network fluctuations, or outages that cause incomplete file downloads.

My solution is to download the dataset directly through a link, and then modify` ImgX-DiffSeg/imgx/datasets/male_ pelvic_ mr/male_ pelvic_ mr_ dataset_ builder.py` and ` amos_ ct_ dataset_ In builder.py`
Change the
```
zip_ dir = dl_ manager.download_ and_ extract(
" https://zenodo.org/record/7013610/files/data.zip "
)
``` 
to 
`zip_ dir = dl_ Manager. extract ('data. zip ') (the path to the downloaded zip file)`

I have tried several methods to directly load the extracted file, but none of them succeeded. 
Who knows how to implement it can share it with us. Can refer to [https://www.tensorflow.org/datasets/add_dataset](url) for details
